sync/recursiveWatcher.go

```diff
@@ -0,0 +1,126 @@
+package sync
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/koyachi/go-term-ansicolor/ansicolor"
+	"gopkg.in/fsnotify.v1"
+)
+
+type RecursiveWatcher struct {
+	*fsnotify.Watcher
+	Files   chan string
+	Folders chan string
+}
+
+func DebugMessage(format string, a ...interface{}) {
+	msg := fmt.Sprintf(format, a...)
+	fmt.Println(ansicolor.IntenseBlack(msg))
+}
+
+func DebugError(msg error) {
+	fmt.Println(ansicolor.IntenseBlack(msg.Error()))
+}
+
+func NewRecursiveWatcher(path string) (*RecursiveWatcher, error) {
+	folders := Subfolders(path)
+	if len(folders) == 0 {
+		return nil, errors.New("No folders to watch.")
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+	rw := &RecursiveWatcher{Watcher: watcher}
+
+	rw.Files = make(chan string, 10)
+	rw.Folders = make(chan string, len(folders))
+
+	for _, folder := range folders {
+		rw.AddFolder(folder)
+	}
+	return rw, nil
+}
+
+func (watcher *RecursiveWatcher) AddFolder(folder string) {
+	err := watcher.Add(folder)
+	if err != nil {
+		log.Println("Error watching: ", folder, err)
+	}
+	watcher.Folders <- folder
+}
+
+func (watcher *RecursiveWatcher) Run(debug bool) {
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				// create a file/directory
+				if event.Op&fsnotify.Create == fsnotify.Create {
```

Comment:
@adragomir I came to learn this is a result of running  `gofmt -e -w ./`
or `make format`